batch.sh

```diff
@@ -3,42 +3,9 @@
 set -u
 set -e
 
-languages=(
-    'bash'
-    'c'
-    'c-sharp'
-    'cmake'
-    'cpp'
-    'css'
-    'dockerfile'
-    'elixir'
-    'glsl'
-    'go'
-    'heex'
-    'html'
-    'java'
-    'javascript'
-    'json'
-    'julia'
-    'make'
-    'markdown'
-    'org'
-    'perl'
-    'proto'
-    'python'
-    'ruby'
-    'rust'
-    'sql'
-    'toml'
-    'tsx'
-    'typescript'
-    'verilog'
-    'vhdl'
-    'wgsl'
-    'yaml'
-)
+if [ -z "${JOBS:-}" ]
+then
+    JOBS=$(nproc) || JOBS=1
+fi
 
-for language in "${languages[@]}"
-do
-    ./build.sh "${language}"
-done
+xargs -r -d '\n' -a "$PWD/languages.txt" -n1 -P"$JOBS" "$PWD/build.sh"
```

Comment:
Note: `-P` is not part of POSIX. It is available on GNU xargs and I'd be amazed to find an xargs implementation that doesn't support it outside of the POSIX specified one. Not sure if this is a concern for you or not.